pkg/driver/node_server.go

```diff
@@ -448,11 +452,15 @@ func (driver *Driver) setupDevice(publishContext map[string]string) (*model.Devi
 		ConnectionMode: defaultConnectionMode,
 	}
 	if publishContext[accessProtocolKey] == iscsi {
-		chapInfo := &model.ChapInfo{
```

Comment:
why to override this here? if CHAP is enabled, then can we make sure its always added in publish context?

Reply:
passwords are not returned from the apis on the array (e.g chap_users), so CSP never returns it, hence the override is needed for password update, username remains the same.
i can modify this to not override if the key doesn't match for other CSPs which return the password.